### PR TITLE
improvement: Support gas estimation for contract functions + overrides object

### PR DESCRIPTION
**Pull request by feliperodriguess**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Enable gas estimation for contract function calls
- Support override options object to be passed to the contract write calls (https://viem.sh/docs/contract/writeContract#gasprice-optional)
- Make options type-safe

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/src/constants/gas-controllers/moe-collection-deploy.ts [hunk 1]
```diff
@@ -139,4 +139,16 @@ export const MOE_COLLECTION_DEPLOY_GAS = {
...
+  [networks.localhost.chain]: {
+    gasPrice: undefined,
+    gasLimit: undefined,
+    gasLimitMultiplier: 120,
+    gasPriceMultiplier: 110,
+  },
+  [networks.localhost2.chain]: {
+    gasPrice: undefined,
+    gasLimit: undefined,
+    gasLimitMultiplier: 120,
```
Comment (alexanderattar) [inline packages/sdk/src/constants/gas-controllers/moe-collection-deploy.ts:151]: Are these values essentially overhead multipliers? in other words, is this like 120%?

**Review by alexanderattar — APPROVED** [review]

File: packages/sdk/src/constants/gas-controllers/moe-collection-deploy.ts (hunk 1 above)
Comment (feliperodriguess) [reply packages/sdk/src/constants/gas-controllers/moe-collection-deploy.ts:151]: yep, i think so

copied from the FE implementation

File: packages/sdk/src/contracts/factory.contract.ts [hunk 2]
```diff
@@ -220,4 +241,38 @@ export class Factory extends HolographBaseContract {
...
+  /**
+   *
+   * @param chainId  The chain id of the network to send the transaction.
+   * @param config The contract deployment configurations.
+   * @param signature The signature by the wallet that created the original payload.
+   * @param signer The address of the signer.
+   * @param wallet Holograph wallet instance, optional param.
+   * @returns The estimated gas for the transaction in wei.
+   */
+  async estimateGasForDeployingHolographableContract(
```
Comment (sogoiii) [inline packages/sdk/src/contracts/factory.contract.ts:264]: @feliperodriguess What is the reason for putting this function in the holograph factory class? My only guess is that this estimate requires uses the factory contract.

Comment (feliperodriguess) [reply packages/sdk/src/contracts/factory.contract.ts:264]: Yes, it requires, and it's tied to the deployHolographableContract, which is a function from the factory contract

Comment (feliperodriguess) [reply packages/sdk/src/contracts/factory.contract.ts:264]: The above is just a wrapper, the final version lives on collection-legacy.ts as you can see on this PR (https://github.com/holographxyz/holograph/pull/32)

Comment (sogoiii) [reply packages/sdk/src/contracts/factory.contract.ts:264]: ok got it.

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**